lib/KettleMiddleware-Multer.js

```diff
@@ -0,0 +1,149 @@
+            fileSize: Infinity,
+            files: Infinity,
+            parts: Infinity,
+            headerPairs: 2000
+        }
+    },
+    middleware: "@expand:kettle.middleware.multer.createMiddleware({that}, {that}.options.middlewareOptions, {that}.getStorage, {that}.getMiddlewareForFileStrategy, {that}.getFileFilter)",
+    invokers: {
+        // Change to getDiskStorage for diskStorage
+        "getStorage": {
+            "func": "{that}.getMemoryStorage"
+        },
+        //
+        "getMiddlewareForFileStrategy": {
+            "funcName": "kettle.middleware.multer.getMiddlewareForFileStrategy",
+            "args": ["{that}", "single", "file"]
+        },
+        // Override this function for different behaviour in the file filter
+        // function
+        "getFileFilter": {
+            "funcName": "kettle.middleware.multer.getAllowAllFileFilter"
+        },
+        "getMemoryStorage": {
+            "funcName": "kettle.npm.multer.memoryStorage",
+            "args": []
+        },
+        "getDiskStorage": {
+            "funcName": "kettle.middleware.multer.getDiskStorage",
+            "args": ["{that}.getDiskStorageDestinationFunc","{that}.getDiskStorageFilenameFunc"]
+        },
+        // Override this function for different behaviour in the destination
+        // when using disk storage
+        "getDiskStorageDestinationFunc": {
+            "funcName": "kettle.middleware.multer.getDefaultDiskStorageDestinationFunc"
+        },
+        // Override this function for different behaviour in the file name
+        // generation when using disk storage
+        "getDiskStorageFilenameFunc": {
+            "funcName": "kettle.middleware.multer.getDefaultDiskStorageFilenameFunc"
+        }
+    }
+});
+
+kettle.middleware.multer.getAllowAllFileFilter = function () {
+    return kettle.npm.multer.allowAll;
+};
+
+kettle.middleware.multer.getMemoryStorage = function () {
+    return kettle.npm.multer.memoryStorage;
+};
+
+// See https://github.com/expressjs/multer#diskstorage for
+// the function style of multer's disk storage destination function
+//
+// If overriding, this should remain a zero-argument function
+// and return an appropriate function for multer's use
+kettle.middleware.multer.getDefaultDiskStorageDestinationFunc = function () {
+    return function (req, file, cb) {
+        cb(null, "./tests/data/uploads");
+    };
+};
+
+// See https://github.com/expressjs/multer#diskstorage for
+// the function style of multer's disk storage file name function
+//
+// If overriding, this should remain a zero-argument function
+// and return an appropriate function for multer's use
+kettle.middleware.multer.getDefaultDiskStorageFilenameFunc = function () {
+    return function (req, file, cb) {
+        cb(null, file.originalname);
+    };
+};
+
+kettle.middleware.multer.getDiskStorage = function (destinationFunc, filenameFunc) {
+    var diskStorage = kettle.npm.multer.diskStorage({
```

Comment:
Since this one is a "thing" it should become a member definition (in an expander) rather than an invoker when unwrapped